robottelo/api/utils.py

```diff
@@ -293,3 +302,242 @@ def create_role_permissions(role, permissions_types_names):  # pragma: no cover
             role=role,
             search=None
         ).create()
+
+
+def configure_provisioning(org=None, loc=None):
+    """Create and configure org, loc, product, repo, cv, env. Update proxy,
+    domain, subnet, compute resource, provision templates and medium with
+    previously created entities and create a hostgroup using all mentioned
+    entities.
+
+    :param org: Default Organization that should be used in both host
+        discovering and host provisioning procedures
+    :param loc: Default Location that should be used in both host
+        discovering and host provisioning procedures
+    :return: List of created entities that can be re-used further in
+        provisioning or validation procedure (e.g. hostgroup or domain)
+    """
+    # Create new organization and location in case they were not passed
+    if org is None:
+        org = entities.Organization().create()
+    if loc is None:
+        loc = entities.Location(organization=[org]).create()
+    # Create a new Life-Cycle environment
+    lc_env = entities.LifecycleEnvironment(organization=org).create()
+    # Create a Product, Repository for custom RHEL6 contents
+    product = entities.Product(organization=org).create()
+    repo = entities.Repository(
+        product=product,
+        url=settings.rhel7_os
+    ).create()
+
+    # Increased timeout value for repo sync
+    try:
+        old_task_timeout = entity_mixins.TASK_TIMEOUT
+        entity_mixins.TASK_TIMEOUT = 3600
+        repo.sync()
+        # Create, Publish and promote CV
+        content_view = entities.ContentView(organization=org).create()
+        content_view.repository = [repo]
+        content_view = content_view.update(['repository'])
+        content_view.publish()
+        content_view = content_view.read()
+        promote(content_view.version[0], lc_env.id)
+    finally:
+        entity_mixins.TASK_TIMEOUT = old_task_timeout
+    # Search for puppet environment and associate location
+    environment = entities.Environment(
+        organization=[org.id]).search()[0].read()
+    environment.location.append(loc)
+    environment = environment.update(['location'])
+
+    # Search for SmartProxy, and associate location
+    proxy = entities.SmartProxy().search(
+        query={
+            u'search': u'name={0}'.format(
+                settings.server.hostname)
+        }
+    )
+    proxy = proxy[0].read()
+    proxy.location.append(loc)
+    proxy = proxy.update(['location'])
+    proxy.organization.append(org)
+    proxy = proxy.update(['organization'])
```

Comment:
I think we can update proxy by one statement like `proxy.update(['location', 'organization'])`